src/cc1101.h

```diff
@@ -139,8 +139,8 @@ namespace cc1101 {
 	#endif
 #endif
 
-	#define wait_Miso()       while(isHigh(misoPin) ) { static uint8_t miso_count=255;delay(1); if(miso_count==0) return; miso_count--; }           // wait until SPI MISO line goes low
-	#define wait_Miso_rf()    while(isHigh(misoPin) ) { static uint8_t miso_count=255;delay(1); if(miso_count==0) return false; miso_count--; }     // wait until SPI MISO line goes low
+	#define wait_Miso()       while(isHigh(misoPin) ) { static uint8_t miso_count; miso_count = 255; delay(1); if(miso_count==0) return; miso_count--; }           // wait until SPI MISO line goes low
```

Comment:
Ich denke dein fix funktioniert so nicht. :-(

`miso_count` wird doch bei jedem Schleifendurchlauf auf 255 gesetzt und wird so niemals 0 erreichen.
Demnach läuft die Schleife endlos, solange `isHigh(misoPin) true  liefert.
So würde man die Endlosschleife verhindern und auch sicherstellen, dass er nie auf 0 stehen bleibt glaube ich:

```
#define wait_Miso_rf()    while(isHigh(misoPin) ) { static uint8_t miso_count=255;delay(1);  miso_count--; if(miso_count==255) return false;}     // wait until SPI MISO line goes low
```

Allerding bringt das ein neues Problemchen mit sich, denn irgendwann steht miso_count mal auf 1, und dann bricht die Schleife vermutlich zu früh ab. Man würde demnach also gerne eine Variable vor dem Start der while schleife definieren und nicht innerhalb der while Schleife.

Reply:
Nein eben nicht. Es ist eine statische Variable.
static heisst hier, es wird genau einmal initialisert - und wenn du wieder an die Stelle kommst, behaelt sie den alten Wert.

https://www.geeksforgeeks.org/static-variables-in-c/

Eigentlich muss die gar nicht static sein. Durch das while ist die dort im Scope und wird beim verlassen der while verworfen. Sie liegt auf dem Stack.

Mein Fix hat deine Initialisierung stehen gelassen (wenn man nichts hinschreiben ist in C und C++ garantiert, dass die bei keyword static mit 0 initialisiert ist - aber das passiert eben genau nur einmal (!).
Was Du danach mit der Variable machst, bleibt erhalten.
Daher 
static uint8t_t miso_count;
Wird miso_couint mit 0 initialisieren.
Danach miso_count = 255 schreibt dort 255 rein. Jedes mal (!)

Also 
static uint8t_t miso_count = 255; ist etwas anderes als static uint8t_t miso_count; miso_count = 255;

Klingt komisch, aber ist so.

Reply:
```
#include <stdio.h>

void functionWithStatic()
{
    static int staticVar = 1;
    printf("%d\n", staticVar);

    staticVar = 2;
}
```

```
$ gcc -Wall -o test_static test_static.c
$ ./test_static 
1
2
```

Weil static, wird dein Init auf 1 im zweiten Run ignoriert, weil ist ja schon initialisiert.

Reply:
Sorry, da fehlt Code:

```
#include <stdio.h>

void functionWithStatic()
{
    static int staticVar = 1;
    printf("%d\n", staticVar);

    staticVar = 2;
}


int main()
{
    functionWithStatic();
    functionWithStatic();
    // wird ausgeben:
    // 1    von der initialisierung
    // 2    initialisierung egal, 2 von der spaeteren zuweisung
    return 0;
}
```